src/lib/kademlia/node_addrs_and_ports.ml

```diff
@@ -8,7 +8,33 @@ type t =
   ; client_port: int
   ; libp2p_port: int
   ; communication_port: int }
-[@@deriving bin_io]
+[@@deriving bin_io, fields]
+
+module Display = struct
+  module Stable = struct
```

Comment:
Would be good to use the versioning ppx here (`module%versioned Stable = struct`).